BigBaseV2/src/script.hpp

```diff
@@ -8,20 +8,24 @@ namespace big
 	public:
 		using func_t = void(*)();
 	public:
-		explicit script(func_t func, std::optional<std::size_t> stack_size = std::nullopt);
+		explicit script(func_t func, const char* name, bool* enabled, std::optional<std::size_t> stack_size = std::nullopt);
 		~script();
 
 		void tick();
 		void yield(std::optional<std::chrono::high_resolution_clock::duration> time = std::nullopt);
 		static script* get_current();
 		static void script_exception_handler(PEXCEPTION_POINTERS exp);
+		const char* m_name;
+		void toggle_script(bool toggle) { if (m_enabled != nullptr) *this->m_enabled = toggle; }
+		bool* get_enabled_pointer() { return m_enabled; }
```

Comment:
Instead of returning the pointer to the bool just dereference the pointer here and return its value.
Additionally you could check if the enabled pointer is valid before dereferencing and return true/false if its a nullptr.

The instead of calling `get_enabled_pointer()` twice each time you need to check if a script is enabled, instead you could just call `is_enabled()`.

```cpp
bool is_enabled() const
{
    if (m_enabled == nullptr)
        return false;
    return *m_enabled;
}
```

Reply:
Ye you could change the default behaviour of the is_enabled or add another member function to get the pointer. My pointer stands that currently you need to make 2 calls for something that can be handled internally by the script member function instead of having duplicate code of checking the pointer each time.

Reply:
As a side note, the boolean can be defaulted within the function so passing a nullptr won't be necessary.

Reply:
> Ye you could change the default behaviour of the is_enabled or add another member function to get the pointer. My point stands that currently you need to make 2 calls for something that can be handled internally by the script member function instead of having duplicate code of checking the pointer each time.

2 calls? A dereference can be employed to both of these things:

`script::get_current()->get_enabled_pointer() //Pointer
*script::get_current()->get_enabled_pointer()//Value`

Reply:
> Ye you could change the default behaviour of the is_enabled or add another member function to get the pointer. My point stands that currently you need to make 2 calls for something that can be handled internally by the script member function instead of having duplicate code of checking the pointer each time.

Thats also the reason for the creation of the vector that only includes scripts that have bools.